File: RouterLib/src/main/java/com/kronos/router/model/RouterParams.java
package com.kronos.router.model;

import java.util.Map;

/**
 * Created by zhangyang on 16/7/16.
 */
public class RouterParams {
    public String url;
    private int weight;
    public RouterOptions routerOptions;
    public Map<String, String> openParams;

    public String getRealPath() {
        try {
            return url.substring(1);
        } catch (Exception e) {
            e.printStackTrace();
        }
        return "";
    }

    public Integer getWeight() {
        return weight;
    }

    public void setWeight(int weight) {
        this.weight = weight;
    }
}

File: RouterLib/src/main/java/com/kronos/router/model/RouterParams.kt
package com.kronos.router.model

/**
 * Created by zhangyang on 16/7/16.
 */
class RouterParams {
    var url: String? = null
    var weight: Int = 0
    var routerOptions: RouterOptions? = null
    var openParams: HashMap<String, String> = hashMapOf()

    val realPath: String
        get() {
            try {
                return url?.let { it.subSequence(1, it.length).toString() } ?: ""
            } catch (e: Exception) {
                e.printStackTrace()
            }
            return ""
        }

    fun put(key: String, value: String?) {
        value?.apply {
            openParams[key] = this
        }
    }
}
